tests/metrics/annotationUtil.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"sync"
+)
+
+//Client request grafana API on a set of resource paths.
+type client struct {
+	// base is the root URL for all invocations of the client
+	baseUrl url.URL
+	client  *http.Client
+}
+
+//Annotation is a specification of the desired behavior of adding annotation
+type Annotation struct {
```

Comment:
Can we add a constructor `NewAnnotation(...)` for better struct defaults?

Reply:
It need more parameters , it is a bad writting.
Actually, tags filed may be empty

Reply:
Typically only required fields are added to constructor parameters, in this case I think the client code shall never omit `Timestamp` and `Text`(an annotation don't have proper message is worthless).
After create the struct, optional fields can be populated by simply assignment, e.g. `anno.DashboardId = 1` or by a setter method if validation logic is required.
Moreover, according to the annotation api, we may add `isRegion` and `timeEnd` fields in the struct, which is helpful when we perform some time consuming operations and want to record both the start and end time.

Reply:
tags field can be empty.
Actually, this object should created from out side, it is not duty for annotation_util class.
Each class should has its own responsibilities